### PR TITLE
Helper Message for no gcm_sender_id in manifest

### PR DESCRIPTION
**Pull request by gauntface**
@wibblymat trying to smooth some parts over while GCM is still in use.

### CODE REVIEW

File: src/client/subscription-failed-error.js [hunk 1]
```diff
@@ -15,7 +15,9 @@ const MESSAGES = {
   'not supported': 'Your browser doesn\'t support push messaging.',
   'denied': 'The user denied permission to show notifications.',
   'dismissed': 'The user dismissed the notification permission dialog.',
-  'endpoint': 'No endpoint URL specified.'
+  'endpoint': 'No endpoint URL specified.',
+  'nogcmid': 'Please ensure you have a Web App Manifest with ' +
+    'a \"gcm_sender_id\" defined.'
```
Comment (wibblymat) [inline src/client/subscription-failed-error.js]: Nit: escaping the double quotes isn't necessary.

File: src/client/push-client.js [hunk 2]
```diff
@@ -111,7 +111,15 @@ export default class PushClient {
       scope: this.scope
     });
     await registrationReady(reg);
-    let sub = await reg.pushManager.subscribe({userVisibleOnly: true});
+    let sub = await reg.pushManager.subscribe({userVisibleOnly: true})
+      .catch((err) => {
+        // This is provide a more helpful message when work with Chrome + GCM
+        var errorToThrow = err;
```
Comment (wibblymat) [inline src/client/push-client.js]: Nit: `var` => `let`

Comment (wibblymat) [inline src/client/push-client.js]: We should add the `no-var` ESLint rule, actually.

Comment (wibblymat) [inline src/client/push-client.js]: I added that rule and pushed it to master.

**Comment by wibblymat** [comment]
Reviewed, just nits. Merge when you're ready :)

File: src/client/push-client.js (hunk 2 above)
Comment (gauntface) [inline src/client/push-client.js]: I'll give it a shot. Only problem I've had with that rule is let isn't
allowed outside of a scope which we may hit.

I'll add it to this PR if that's cool with you.

**Pull request merged**